src/components/Application/Navigation.tsx

```diff
@@ -0,0 +1,133 @@
+
+  const nextSection = (): SectionDetails | null => {
+    const nextSections = Object.values(sections)
+      .filter(({ details: { index } }) => index > currentSectionDetails.index)
+      .sort(({ details: { index: aIndex } }, { details: { index: bIndex } }) => aIndex - bIndex)
+    return nextSections.length > 0 ? nextSections[0].details : null
+  }
+
+  const previousSection = (): SectionDetails | null => {
+    const previousSections = Object.values(sections)
+      .filter(({ details: { index } }) => index < currentSectionDetails.index)
+      .sort(({ details: { index: aIndex } }, { details: { index: bIndex } }) => bIndex - aIndex)
+    return previousSections.length > 0 ? previousSections[0].details : null
+  }
+
+  const isFirstPage = current.pageNumber - 1 === 0 && previousSection() == null
+  const isLastPage =
+    current.pageNumber + 1 > currentSectionDetails.totalPages && nextSection() == null
+
+  const getPreviousSectionPage = (): SectionAndPage => {
+    const { sectionCode, pageNumber } = current
+    if (pageNumber > 1) return { sectionCode, pageNumber: pageNumber - 1 }
+    return {
+      sectionCode: (previousSection() as SectionDetails).code,
+      pageNumber: (previousSection() as SectionDetails).totalPages,
+    }
+  }
+
+  const getNextSectionPage = (): SectionAndPage => {
+    const { sectionCode, pageNumber } = current
+    if (pageNumber < currentSectionDetails.totalPages)
+      return { sectionCode, pageNumber: pageNumber + 1 }
+    return {
+      sectionCode: (nextSection() as SectionDetails).code,
+      pageNumber: 1,
+    }
+  }
+
+  const sendToPage = (sectionPage: SectionAndPage) => {
+    const { sectionCode, pageNumber } = sectionPage
+    push(`/applicationNEW/${serialNumber}/${sectionCode}/Page${pageNumber}`)
+  }
+
+  const previousButtonHandler = (_: any) => {
+    const previousSectionPage = getPreviousSectionPage()
+    sendToPage(previousSectionPage)
+  }
+
+  const nextPageButtonHandler = (_: any) => {
+    const nextSectionPage = getNextSectionPage()
+    // Use validationMethod to check if can change to page (on linear application) OR
+    // display current page with strict validation
+    requestRevalidation(({ firstStrictInvalidPage, setStrictSectionPage }: MethodToCallProps) => {
+      if (
+        firstStrictInvalidPage !== null &&
+        current.sectionCode === firstStrictInvalidPage.sectionCode &&
+        current.pageNumber === firstStrictInvalidPage.pageNumber
+      ) {
+        setStrictSectionPage(firstStrictInvalidPage)
+      } else {
+        setStrictSectionPage(null)
+        sendToPage(nextSectionPage)
+      }
+    })
+  }
+
+  return (
+    <Sticky
+      pushing
+      style={{ backgroundColor: 'white', boxShadow: ' 0px -5px 8px 0px rgba(0,0,0,0.1)' }}
+    >
+      <Segment.Group horizontal>
+        <Segment style={{ minWidth: '50%' }}>
+          {!isFirstPage && (
+            <Button
+              basic
```

Comment:
I think we should avoid using `float` for layout most of the time. It's an old-fashioned way of doing it before Flexbox was a thing, and things that are floated right don't wrap properly on mobile:
![Screen Shot 2021-03-04 at 12 59 25 PM](https://user-images.githubusercontent.com/5456533/109889118-a2791300-7ce9-11eb-89ef-c9960df9d34b.png)

I think you have to use Grid to lay this out properly, or maybe manually do flex box. Have a look at the Pagination component for one way of doing it (Yes, I know there is a `floated` in there, but it's actually no longer needed -- it looks the same if you remove the floats)

Reply:
Yea, good point! Will have a look after the meeting : )

Reply:
Hi! I have looked in `Pagination` and changed to use Grid, but I can still see some not beautiful UI on mobile... I'm calling out for a good to have in App layout issue. But have changed to be using Grid, so hopefully better than before.

On mobile:
<img width="325" alt="Screen Shot 2021-03-04 at 3 47 26 PM" src="https://user-images.githubusercontent.com/16461988/109903867-4326fd00-7d01-11eb-87f2-62e66bfa643d.png">

On desktop:
<img width="728" alt="Screen Shot 2021-03-04 at 3 47 38 PM" src="https://user-images.githubusercontent.com/16461988/109903859-40c4a300-7d01-11eb-90bd-4d21866802ef.png">